altair/vegalite/v2/api.py

```diff
@@ -164,8 +190,18 @@ def encode(self, *args, **kwargs):
                 # as part of the to_dict() call.
                 kwargs[prop] = cls.from_dict(field, validate=False)
         # TODO: update nested values rather than overwriting them
-        self.encoding = core.EncodingWithFacet(**kwargs)
-        return self
+        copy = self.copy(deep=True, ignore=['data'])
+        encoding = copy.encoding
+        if encoding is Undefined:
+            encoding = {}
+        elif isinstance(encoding, dict):
```

Comment:
Doesn't this mean that if the encoding already exists, you can't call encode multiple times? I could imagine that would be useful to do some encodings in a base chart and then add others in combinations with hconcat/vconcat/layer.

Reply:
Also, if it is a dict, wouldn't we want to wrap it in an `EncodingWithFacet` object?

Reply:
So what's going on here is that if copy.encoding is a dict, it's already been deep-copied and so we don't need to do anything. So what you have in mind is possible.

I think it's a good idea to wrap it; I'll make that change.

Reply:
Actually, now that I look at it, this dict is wrapped with an ``EncodingWithFacet`` object a few lines later, so it's already good. I've added some clarifying comments.